src/Jackalope/NodeType/NodeProcessor.php

```diff
@@ -343,19 +344,23 @@ private function validateNamespace($name)
     {
         $alias = null;
         $aliasLength = strpos($name, ':');
-        if (false === $aliasLength) {
+
+        if ($aliasLength === false) {
             return;
-        } 
-        
-        $alias = substr($name, 0, $aliasLength);
+        } else {
+            $alias = substr($name, 0, $aliasLength);
+        }
 
         if (!isset($this->namespaces[$alias])) {
-            throw new NamespaceException(sprintf(
-                'Namespace alias "%s" is not known for name "%s", known namespace aliases are: "%s"',
-                $alias,
-                $name,
-                implode('", "', array_keys($this->namespaces))
-            ));
+            if ($this->namespaces instanceof ArrayObject) {
+                $this->namespaces = $this->namespaces->getArrayCopy();
```

Comment:
Yes, I thought over this line... but I wanted to know your feedback :) I will update it :)